generics/services/users.js

```diff
@@ -224,9 +226,113 @@ const profileReadPrivate = function (userId) {
+
+/**
+  * get subEntities of matching type by recursion.
+  * @method
+  * @name getSubEntitiesBasedOnEntityType
+  * @param parentIds {Array} - Array of entity Ids- for which we are finding sub entities of given entityType
+  * @param entityType {string} - EntityType.
+  * @returns {Array} - Sub entities matching the type .
+*/
+
+async function getSubEntitiesBasedOnEntityType( parentIds, entityType, result ) {
+
+    if( !parentIds.length > 0 ){
+        return result;
+    }
+    let bodyData={
+        "parentId" : parentIds
+    };
+
+    let entityDetails = await locationSearch(bodyData);
+    if( !entityDetails.success ) {
+        return (result);
+    }
+
+    let entityData = entityDetails.data;
+    let parentEntities = [];
+    entityData.map(entity => {
+    if( entity.type == entityType ) {
+        result.push(entity.id)
+    } else {
+        parentEntities.push(entity.id)
+    }
+    });
+    
+    if( parentEntities.length > 0 ){
+        await getSubEntitiesBasedOnEntityType(parentEntities,entityType,result)
+    } 
+    
+    let uniqueEntities = _.uniq(result);
+    return uniqueEntities;    
+}
+
+
+/**
+  * get user roles data token.
+  * @method
+  * @name getUserRoles
+  * @param {Object} roles - {"roles":"all"} 
+  * @returns {Array} - All user roles
+*/
+const getUserRoles = function (
+    filterData =  "all",
+    projection = "all",
+    skipFields = "none"
+) {
+    return new Promise(async (resolve, reject) => {
+        try {
+            let url = userServiceUrl + CONSTANTS.endpoints.LIST_USER_ROLES 
+            const options = {
+                headers : {
+                    "content-type": "application/json",
+                    "internal-access-token": process.env.INTERNAL_ACCESS_TOKEN,
+                },
+                json : {
+                    query : filterData,
+                    projection : projection,
+                    skipFields : skipFields
+                }
+            };
+            request.get(url,options,requestCallback);
+            let result = {
+                success : true
+            };
+            function requestCallback(err, data) { 
+                if (err) {
+                    result.success = false;
+                } else {
+                    let response = JSON.parse(data.body);
+                    if( response.responseCode === HTTP_STATUS_CODE.ok.code ) {
+                        result["data"] = response.result;
+                    } else {
+                        result.success = false;
+                    }
+
+                }   
+                return resolve(result);
+            }
+            setTimeout(function () {
+                return resolve (result = {
+                    success : false
+                 });
+             }, CONSTANTS.common.SERVER_TIME_OUT);
+
+        } catch (error) {
+            return reject(error);
+        }
+    })
+}
+
+
 module.exports = {
     profile : profile,
     locationSearch : locationSearch,
```

Comment:
@Prajwal17Tunerlabs if we are not using these functions please make sure we are not calling it anywhere.

Reply:
i have commented the unused code for now